packages/dx-scheduler-core/src/plugins/editing-state/helpers.ts

```diff
@@ -0,0 +1,220 @@
+import moment from 'moment';
+import { RRule, rrulestr, RRuleSet } from 'rrule';
+import {
+  AppointmentModel, PreCommitChangesFn, Changes, MakeDateSequenceFn, EditFn, DeleteFn,
+} from '../../types';
+import { RECURRENCE } from '../../constants';
+
+const mergeNewChanges = (
+  appointmentData: Partial<AppointmentModel>, changes: Changes,
+) => {
+  const appointment = {
+    ...appointmentData,
+  };
+  delete appointment.id;
+  delete appointment.rRule;
+  delete appointment.exDate;
+  delete appointment.parentData;
+  return { ...appointment, ...changes };
+};
+
+const reduceExDate = (prevExDate: string, boundDate: Date) => {
+  if (prevExDate.length > 0) {
+    return prevExDate.split(',').reduce((acc: string[], date: string) => {
+      const momentDate = moment.utc(date);
+      if (momentDate.isBefore(boundDate)) {
+        return [...acc, date];
+      }
+      return acc;
+    }, []).join(',');
+  }
+  return undefined;
+};
+
+const configureExDate = (exDate: string | undefined, date: Date) => {
+  const currentExDate = `${moment.utc(date)
```

Comment:
It's better to place it on a single line or to not use template string here because currentExDate contains uneeded newline